WarshipGame/Assets/Scripts/AI/Enemy.cs

```diff
@@ -22,6 +27,8 @@ public class Enemy : MonoBehaviour
     [Header("Movement")]
     [SerializeField] private Waypoint WaypointsScript;
 
+    private int _waterLayer = 4;
```

Comment:
_waterLayer is redundant as it is only used in OnCollisionEnter() and in OnCollisionExit() to check wheter something is not equal to _waterLayer which is always 4

Reply:
true frfr